### ISSUE TITLE
Service not able to find other service after update

### ISSUE THREAD

**Issue by otaviosoares** [senecajs/seneca-mesh#48]
Sometimes I'm getting this weird behavior. It doesn't happen all the time, so it's very difficult to reproduce and solve. So, I'd like to know if anyone has clues on it.

**Scenario:**
Service A is running.
Service B is running.
Service B uses service A.

**Steps:**
Service B is updated.
Service A is running.
Service B doesn't find Service A anymore (No matching action pattern found for ......)

I restart service A and B
Everything works again

seneca-mesh version: 0.8 (Also tested with prior versions)

Any help would be appreciated.

Thanks

**Comment by StarpTech** [comment]
@otaviosoares  Can you write a test and create a PR ?

**Comment by otaviosoares** [comment]
@StarpTech I'd love to. The problem is I can't reproduce it locally. It just happens randomly in my production env. I'm not sure if it's a bug or I'm doing something wrong.

**Comment by StarpTech** [comment]
@otaviosoares This scenario is hard to debug. Can you post you configs and explain the services more precise?

**Comment by otaviosoares** [comment]
Sure. Here's a piece of my configs: https://gist.github.com/otaviosoares/36266af6a34f821eef843a203749fe84

As you can see, it's not a big deal. However, there are some things that worth the mention.

It r

**Comment by otaviosoares** [comment]
Sure. Here's a piece of my configs: https://gist.github.com/otaviosoares/36266af6a34f821eef843a203749fe84

As you can see, it's not a big deal. However, there are some things that worth the mention.

- It's running in a Kubernetes Cluster.
- The Base IP is automatically injected in an env variable - https://gist.github.com/otaviosoares/36266af6a34f821eef843a203749fe84#file-1_auth-js-L6
- The Service Host Ip is also automatically injected - https://gist.github.com/otaviosoares/36266af6a34f821eef843a203749fe84#file-1_auth-js-L5
- When the error occurs, I connect to the base container and manually run the role:mesh,get:members. It returns fewer entries than running services, it's like it's losing the reference to the services
- If I restart the service it shows on the base get:members.

Please, let me know if it's not detailed enough.
Thanks

**Comment by otaviosoares** [comment]
Just to give this issue a closure.

I was able to find what was causing the error.

My base service had no reference to other bases. So, if I restarted one of my bases for any reason, it had no references to other services. It started a cascade effect, any service using this base to join the network would have no references as well.

Solution: Started using consul to registry bases instead of defining it myself.

Thank you
